shared/constants/signup.tsx

```diff
@@ -8,8 +8,11 @@ export const usernameHint =
 export const noEmail = 'NOEMAIL'
 export const waitingKey = 'signup:waiting'
 
-const defaultDevicename =
-  Platforms.realDeviceName ||
+// Remove parens from the device name. Keybase device names cannot have parens.
+const cleanupDeviceName = (name: string) => name.replace(/[()]/g, '')
```

Comment:
this should be more expansive

Reply:
On the client here are the rules
- must match `^[a-zA-Z0-9][ _'a-zA-Z0-9+‘’—–-]*$`
- 3 <= len <= 64
- some stuff with curly quotes
- must not match `  |[ '_-]$|['_-][ ]?['_-]` (github trims spaces on this regex)

How expansive? The error if you just hit the blue button is pretty clear about the rules. If it messes with these rules then drop it and use the default?

Reply:
just replace all the non a-Z-_

Reply:
My ipad is named `Miles's iPad` which is valid but that would turn it into `Miless iPad`